### PR TITLE
(closes #1505) fix bug where kind parameter of UnknownFortranType was replaced

### LINKED ISSUE #1505: [psyir] Parameter initialisation lost when initialisation expression results in a CodeBlock
The following declaration:

    INTEGER, PRIVATE, PARAMETER ::       &  !: Floating point section
      sp = SELECTED_REAL_KIND( 6, 37),  &  !:  Single precision (real 4)
      dp = SELECTED_REAL_KIND(12,307),  &  !:  Double precision (real 8)
      wp = dp                              !:  Working precision

is reproduced by PSyclone as:

    integer, private, parameter :: sp = SELECTED_REAL_KIND(6, 37)
    integer, private, parameter :: dp = SELECTED_REAL_KIND(12, 307)
    integer, private :: wp

i.e. `wp` is no longer given a value (and isn't declared as a parameter either).

### PR DESCRIPTION
**Pull request by arporter**

### CODE REVIEW

**Comment by arporter** [comment]
Assuming CI happiness, this is ready for first review. One for either @sergisiso or @rupertford.

File: src/psyclone/tests/psyir/frontend/fparser2_kind_params_test.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors R. W. Ford, A. R. Porter and S. Siso, STFC Daresbury Lab
+# Modified I. Kavcic, Met Office
+# -----------------------------------------------------------------------------
+
+''' Performs py.test tests on the support for KIND parameters in the
+    fparser2 PSyIR front-end '''
+
+from __future__ import absolute_import
```
Comment (sergisiso) [inline src/psyclone/tests/psyir/frontend/fparser2_kind_params_test.py:41]: This import is no longer needed

File: src/psyclone/psyir/symbols/datasymbol.py [hunk 2]
```diff
@@ -144,8 +144,11 @@ def constant_value(self, new_value):
             instance, or 5) the provided PSyIR expression is unsupported.
 
         '''
-        from psyclone.psyir.nodes import Node, Literal, Operation, Reference
-        from psyclone.psyir.symbols import ScalarType, ArrayType
+        # pylint: disable=import-outside-toplevel
```
Comment (sergisiso) [inline src/psyclone/psyir/symbols/datasymbol.py:147]: Update copyright date

File: src/psyclone/tests/psyir/symbols/datasymbol_test.py [hunk 3]
```diff
@@ -41,12 +41,16 @@
 from __future__ import absolute_import
 import pytest
 
+from fparser.common.readfortran import FortranStringReader
```
Comment (sergisiso) [inline src/psyclone/tests/psyir/symbols/datasymbol_test.py:44]: Update copyright date and remove absolute_import

**Review by sergisiso — CHANGES_REQUESTED** [review]
@arporter Well done, I just made some very minor comments. Compilation test and examples pas, no remaining todos, fixed the touched 'disable_declaration_check' (thanks!), reference guide still build.

File: src/psyclone/tests/psyir/frontend/fparser2_kind_params_test.py (hunk 1 above)
Comment (arporter) [reply src/psyclone/tests/psyir/frontend/fparser2_kind_params_test.py:41]: Oh yes, thanks.

File: src/psyclone/psyir/symbols/datasymbol.py (hunk 2 above)
Comment (arporter) [reply src/psyclone/psyir/symbols/datasymbol.py:147]: Oops/

File: src/psyclone/tests/psyir/symbols/datasymbol_test.py (hunk 3 above)
Comment (arporter) [reply src/psyclone/tests/psyir/symbols/datasymbol_test.py:44]: Done.

**Comment by arporter** [comment]
Assuming CI happiness, this is ready for another look.

**Review by sergisiso — APPROVED** [review]
@arporter All comments have been addressed. Test pass, no documentation is needed. Formatters are ok. Reference guide builds. No remaining references to the issues. This is ready to merge.

**Pull request merged**